### DISCUSSION TITLE
AWS Lambda Function Bridges for chatwoot integrations

### DISCUSSION THREAD

**Discussion by thundersparkf** [chatwoot/chatwoot#4293] [Show and tell]
Greetings chatwoot,

As a part of @weunlearn's solution platform for adolescent kids helping them better understand gender and mental health, we've used Chatwoot extensively as way to monitor interactions with the kids. Over the year, we've had to integrate it with tools like [Gupshup](https://www.gupshup.io), [Rasa](https://rasa.com) and [Botpress](https://botpress.com), and this has led to us developing AWS Lambda functions to serve as bridges. As of now, we have:

Rasa agent bot AWS Lambda bridge: NodeJS

Gupshup to Chatwoot API AWS Lambda bridge: NodeJS

Chatwoot to Gupshup API AWS Lambda bridge: NodeJS

Botpress agent bot AWS Lambda bridge(with AWS translator service): GoLang

If this is something that chatwoot is interested in, we'd be happy to share our code so other users can leverage these functionalities in production efficiently.

**Comment by sojan-official** [comment]
@thundersparkf  This would be amazing if you can share these extensions with the community.  We can add documentation for these integrations on https://github.com/chatwoot/docs with links to the repo/repos. 
or have a dedicated repo in chatwoot org for community integrations. let me know how you would like to proceed.

**Reply by thundersparkf** [reply]
Currently our repos are in GitLab so we'll migrate those here and we'll add you to those. You can review those and let us know your thoughts about what to do next. Does this sound good?

**Reply by sojan-official** [reply]
@thundersparkf Thanks a lot. That sounds good.

**Reply by sojan-official** [reply]
@thundersparkf I have added a new repo: https://github.com/chatwoot/implementation-examples 

- if migrating from Gitlab is too much effort / the original repos are open. Then we can add a new folder for each implementation to these repos with a `readme.md` with a link and instructions to the original repo. 
- or we could maintain a folder for each implementation in this newly created repo.

**Reply by thundersparkf** [reply]
No worries @sojan-official we've been meaning to do it as well. I've migrated the gitlab repo for botpress lambda function. [Chatwoot Botpress Lambda](https://github.com/weunlearn/Chatwoot-Botpress-Bridge.git). Lemme know what will be the next step. Also, let's move this to discord if that's okay with you

**Reply by sojan-official** [reply]
@thundersparkf sounds good. let's take it forward on discord. I will link the Botpress repo to the implementation example one.

**Reply by sojan-official** [reply]
linked the repo: https://github.com/chatwoot/implementation-examples/tree/main/chatwoot-botpress-bridge
